### ISSUE TITLE
DropdownMenu scrolls unnecessarily when the user types something into it

### ISSUE THREAD

**Issue by Hixie** [flutter/flutter#151854]
When `enabledSearch` is enabled (the default), if a DropdownMenu is in a scrollable, then when you type some text the scrollable jumps. Sometimes it jumps enough to move the DropdownMenu out of view (!). It's not clear why it needs to scroll at all. The widget works basically fine when you turn `enabledSearch` off (except for lacking search features and some minor other bugs), despite not scrolling. Should the scrolling logic just be removed?

This makes the widget basically unusable for text input.

(Testing on main, 3.24.0-1.0.pre.104, on macOS 14.2.1 23C71 darwin-arm64.)
